packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.test.tsx

```diff
@@ -0,0 +1,233 @@
+            username: 'kobe@test.com',
+            password: 'Test1234!'
+        })
+    }, [])
+
+    return (
+        <>
+            {loginRegisteredUser.isLoading ? (
+                <span>Logging in...</span>
+            ) : (
+                <div>Logged in as {loginRegisteredUser?.variables?.username}</div>
+            )}
+            <div>
+                <button
+                    onClick={() =>
+                        updateCustomer.mutate({
+                            parameters: {customerId: CUSTOMER_ID},
+                            body: {firstName: `Kobe${RANDOM_STR}`}
+                        })
+                    }
+                >
+                    Update Customer
+                </button>
+
+                {updateCustomer.error?.message && (
+                    <p style={{color: 'red'}}>Error: {updateCustomer.error?.message}</p>
+                )}
+                <hr />
+                <div>{JSON.stringify(updateCustomer)}</div>
+            </div>
+        </>
+    )
+}
+
+const tests = [
+    {
+        hook: 'updateCustomer',
+        cases: [
+            {
+                name: 'success',
+                assertions: withMocks(async () => {
+                    renderWithProviders(<CustomerMutationComponent />)
+                    await waitFor(() => screen.getByText(/kobe@test.com/))
+                    await waitFor(() =>
+                        screen.getByRole('button', {
+                            name: /update customer/i
+                        })
+                    )
+
+                    // CACHE Pre-populate cache with the query keys we invalidate on success
+                    queryClient.setQueryData(
+                        ['/customers', CUSTOMER_ID, '/payment-instruments'],
+                        {}
+                    )
+                    queryClient.setQueryData(['/customers', CUSTOMER_ID, '/addresses'], {})
+                    queryClient.setQueryData(['/customers', '/external-profile'], {})
+
+                    expect(
+                        queryClient.getQueryState([
+                            '/customers',
+                            CUSTOMER_ID,
+                            '/payment-instruments'
+                        ])?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', CUSTOMER_ID, '/addresses'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+                    expect(
+                        queryClient.getQueryState(['/customers', '/external-profile'])
+                            ?.isInvalidated
+                    ).toBeFalsy()
+
+                    // Mocking the server request
+                    nock('http://localhost:3000')
+                        .patch((uri) => {
+                            return uri.includes('/customer/shopper-customers/')
+                        })
```

Comment:
I think this manual `nock` is necessary because we're using a different random string each time this test suite is ran.

If we don't have the random strings, we can remove this nock and rely on `nockBack` to record the http responses instead.

Reply:
The usage of `nock` here  is to establish the pattern of capturing PATCH requests made when executing mutation hooks and preserve the state of the server, without the need of doing a real request to the server or recording a real http response.